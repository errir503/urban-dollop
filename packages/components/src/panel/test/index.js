/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import Panel from '../';

describe( 'Panel', () => {
	describe( 'basic rendering', () => {
		it( 'should render an empty div without any provided props', () => {
			const panel = shallow( <Panel /> );
			expect( panel.hasClass( 'components-panel' ) ).toBe( true );
			expect( panel.type() ).toBe( 'div' );
			expect( panel.find( 'div' ).shallow().children() ).toHaveLength(
				0
			);
		} );

		it( 'should render a PanelHeader component when provided text in the header prop', () => {
			const panel = shallow( <Panel header="Header Label" /> );
			const panelHeader = panel.find( 'PanelHeader' );
			expect( panelHeader.prop( 'label' ) ).toBe( 'Header Label' );
			expect( panel.find( 'div' ).shallow().children() ).toHaveLength(
				1
			);
		} );

		it( 'should render an additional className', () => {
			const panel = shallow( <Panel className="the-panel" /> );
			expect( panel.hasClass( 'the-panel' ) ).toBe( true );
		} );

		it( 'should add additional child elements to be rendered in the panel', () => {
			const panel = shallow( <Panel children="The Panel" /> );
			expect( panel.text() ).toBe( 'The Panel' );
			expect( panel.find( 'div' ).shallow().children() ).toHaveLength(
				1
			);
		} );

		it( 'should render both children and header when provided as props', () => {
			const panel = shallow(
				<Panel children="The Panel" header="The Header" />
			);
			expect( panel.find( 'div' ).shallow().children() ).toHaveLength(
				2
			);
		} );
	} );
} );
