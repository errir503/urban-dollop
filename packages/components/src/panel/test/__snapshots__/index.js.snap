// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel basic rendering should render an additional className 1`] = `
<div>
  <div
    class="the-panel components-panel"
  />
</div>
`;

exports[`Panel basic rendering should render an empty div without any provided props 1`] = `
<div>
  <div
    class="components-panel"
  />
</div>
`;
